### PR TITLE
Strip choice number readout from NCs on dev server

### PR DESCRIPTION
**Pull request by gausie**
You'll have to take my word for the test HTML! devsters get a readout of the choice number when visiting a choice, which mafia records as part of lastEncounter. It's convenient for me to automate testing in mafia just as much as it's convenient for me to automate playing, and stripping this makes it more compatible with various library scritps I use that check the string value of lastEncounter

### CODE REVIEW

File: src/net/sourceforge/kolmafia/request/AdventureRequest.java [hunk 1]
```diff
@@ -390,7 +405,8 @@ public static final String registerEncounter(final GenericRequest request) {
           };
     }
 
-    String prettyEncounter = StringUtilities.getEntityDecode(encounter);
+    String prettyEncounter = stripDevReadout(StringUtilities.getEntityDecode(encounter));
```
Comment (jaadams5) [inline src/net/sourceforge/kolmafia/request/AdventureRequest.java:408]: Conceptually I would rather only do this if mafia knows it is running on the dev server.  But there are very few (two?) places where mafia knows that and, in the absence of a profiler, I can certainly understand that checking might be as expensive as just running stripDevReadout unconditionally.  A comment, not a blocker.

**Review by jaadams5 — APPROVED** [review]
I expect the number of users who will benefit can be counted on one hand but given who those users are, I'm OK with this.

File: src/net/sourceforge/kolmafia/request/AdventureRequest.java (hunk 1 above)
Comment (gausie) [reply src/net/sourceforge/kolmafia/request/AdventureRequest.java:408]: I can see if it's easy to check for that!

**Review by midgleyc — CHANGES_REQUESTED** [review]
Both `useDevProxyServer` and `loginServerName` should be able to be used to check whether you're on the dev server or not.

Can you check before calling `stripDevReadout`?

**Comment by gausie** [comment]
> Both `useDevProxyServer` and `loginServerName` should be able to be used to check whether you're on the dev server or not.
> 
> Can you check before calling `stripDevReadout`?

@midgleyc done

**Review by midgleyc — APPROVED** [review]
Thanks!

**Pull request merged**